modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -213,18 +215,20 @@ private void resolveTxStateFromTxCoordinator(
         updateLocalTxMapAfterDistributedStateResolved(txId, txMetaFuture);
 
         InternalClusterNode coordinator = coordinatorId == null ? null : clusterNodeResolver.getById(coordinatorId);
-
         if (coordinator == null) {
             // This means the coordinator node have either left the cluster or restarted.
             markAbandoned(txId);
 
             resolveTxStateFromCommitPartition(txId, commitGrpId, txMetaFuture);
         } else {
-            txMessageSender.resolveTxStateFromCoordinator(coordinator.name(), txId, timestamp)
+            txMessageSender.resolveTxStateFromCoordinator(coordinator, txId, timestamp)
                     .whenComplete((response, e) -> {
-                        if (e == null) {
-                            txMetaFuture.complete(asTransactionMeta(response.txStateMeta()));
+                        if (e == null && response.txStateMeta() != null) {
+                            txMetaFuture.complete(asTransactionMeta(Objects.requireNonNull(response.txStateMeta())));
```

Comment:
It's better to extract `Objects.requireNonNull` and add a message to the verification.